plugins/clipboard/plugin.js

```diff
@@ -2539,6 +2583,227 @@
+	 * for browsers supporting Clipboard API, but not supporting custom MIME types (Edge 16+).
+	 *
+	 * @since 4.8
+	 * @class CKEDITOR.plugins.clipboard.fallbackDataTransfer
+	 * @constructor Creates a class instance.
+	 * @param {Object} [nativeDataTransfer] A native data transfer object.
+	 * @param {String} [customDataFallbackType=text/html] A MIME type used for storing custom data types.
+	 */
+	CKEDITOR.plugins.clipboard.fallbackDataTransfer = function( nativeDataTransfer, customDataFallbackType ) {
+
+		/**
+		 * A native dataTransfer object.
+		 *
+		 * @private
+		 * @property {Object} _nativeDataTransfer
+		 */
+		this._nativeDataTransfer = nativeDataTransfer;
+
+		/**
+		 * A MIME type used for storing custom MIME types. Defaults to 'text/html'.
+		 *
+		 * @private
+		 * @property {String} _customDataFallbackType
+		 */
+		this._customDataFallbackType =  customDataFallbackType || 'text/html';
+	};
+
+	/**
+	 * True if the environment supports custom MIME types in `dataTransfer/cliboardData` `getData/setData` methods. Introduced
+	 * to distinguish browsers which supports only some whitelisted types (like 'text/html', 'application/xml'), but does not support
+	 * custom MIME types (like `cke/id`) like Edge 16+.
+	 * This property should not be accessed directly, use {@link CKEDITOR.plugins.clipboard.fallbackDataTransfer#isRequired}
+	 * method instead.
+	 *
+	 * @private
+	 * @static
+	 * @property {Boolean}
+	 */
+	CKEDITOR.plugins.clipboard.fallbackDataTransfer._isCustomMimeTypeSupported = null;
+
+	CKEDITOR.plugins.clipboard.fallbackDataTransfer.prototype = {
+		/**
+		 * Whether {@link CKEDITOR.plugins.clipboard.fallbackDataTransfer} should be used when operating on dataTransfer.
+		 * If true is returned, it means custom MIME types are not supported in the current browser
+		 * (see {@link CKEDITOR.plugins.clipboard.fallbackDataTransfer#_isCustomMimeTypeSupported}).
+		 *
+		 * @returns {Boolean}
+		 */
+		isRequired: function() {
+			// If there is no `dataTransfer` we cannot detect if fallback is needed.
+			// Method returns `false` so regular flow will be applied.
+			if ( !this._nativeDataTransfer ) {
+				return false;
+			}
+
+			if ( CKEDITOR.plugins.clipboard.fallbackDataTransfer._isCustomMimeTypeSupported === null ) {
+
+				CKEDITOR.plugins.clipboard.fallbackDataTransfer._isCustomMimeTypeSupported = true;
+
+				try {
+					this._nativeDataTransfer.setData( 'cke/mimetypetest', 'test value' );
+					this._nativeDataTransfer.setData( 'cke/mimetypetest', null );
+				} catch ( e ) {
+					if ( this._isUnsupportedMimeTypeError( e ) ) {
+						CKEDITOR.plugins.clipboard.fallbackDataTransfer._isCustomMimeTypeSupported = false;
+					}
+				}
+			}
+
+			return !CKEDITOR.plugins.clipboard.fallbackDataTransfer._isCustomMimeTypeSupported;
+		},
+
+		/**
+		 * Returns the data of the given MIME type. Extracts data of the given MIME type if it is stored in
```

Comment:
Not sure if `removes` is the proper word as it may indicate that the special comment is removed from the stored data which is not true. The intention of this description (and the method itself) is to:
 _Return the data of the given MIME type if it is stored in normal way or in the special comment. Also if the given MIME type is the same as `_customDataFallbackType`, the special comment is removed from returned data_.

☝️ Maybe we should describe it in the similar manner in specs?

Reply:
The above sounds much better, though still I think we're making it a bit too complex 😞. I think we'll need a hand from @AnnaTomanek. @AnnaTomanek could you have a look at it now, or do you prefer to do it after the merge to the major branch?